dev/PushNotifications/PushNotificationsLongRunningTask/ToastRegistrationManager.h

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <unordered_map>
+#include <algorithm>
+#include <vector>
+
+class ToastRegistrationManager
+{
+public:
+    ToastRegistrationManager();
+
+    void Add(std::wstring const& processName, std::wstring const& toastAppId);
+
+    void Remove(std::wstring const& processName);
+
+    std::wstring GetToastRegistration(std::wstring const& processName);
+
+private:
+    winrt::Windows::Storage::ApplicationDataContainer m_toastStorage{ nullptr };
+
+    std::unordered_map<std::wstring, std::wstring> m_registrationMap = {};
```

Comment:
>p<std::wstring,  [](http://example.com/codeflow?start=21&length=16)

Are the process names' casing important? map's default hash/comparator is case-sensitive.

Reply:
Yes, processName is case-sensitive. We assume the processName we get from GetCurrentProcessPath is consistent.

Reply:
Should we always change processName that is returned as part of our Process APIs to lower case so that we don't hit compare issues?

Reply:
I guess my question would be how would the casing change from run-to-run? It's probably better to be defensive and use all lowercase.

Reply:
Fixed but should look into adding comparator to map.

Issue to track: https://github.com/microsoft/WindowsAppSDK/issues/2225